tower-http/src/compression/compression_filter.rs

```diff
@@ -0,0 +1,26 @@
+use crate::compression_utils::default_compression_filter_predicate;
+
+/// A filter which any response Parts needs to pass to be compressed
+pub trait CompressionFilter: Copy {
+    /// Predicate which takes response parts and returns true if the response should be compressed
+    fn should_compress(&self, parts: &http::response::Parts) -> bool;
```

Comment:
I know it was me who suggested using `request::Parts` in the first place but we talked about it in https://github.com/tower-rs/tower-http/issues/154 and decided to stick with `Request`. So I think we should do that here as well to remain consistent.

```suggestion
    fn should_compress<B>(&self, request: &Request<B>) -> bool;
```